lib/Sema/TypeCheckDeclOverride.cpp

```diff
@@ -595,6 +595,106 @@ static bool parameterTypesMatch(const ValueDecl *derivedDecl,
   return true;
 }
 
+/// Returns true if `derivedDecl` has a `@differentiable` attribute that
+/// overrides one from `baseDecl`.
+static bool hasOverridingDifferentiableAttribute(ValueDecl *derivedDecl,
```

Comment:
Similar to my comment below about protocol requirements, it might be good to exclude this from this PR.

Reply:
I'm open to splitting this PR, but to save time, I won't do so preemptively. I'll cc a code owner soon, we can split if requested.